__tests__/components/CartContent/EditPopUp/EditCartItemPopUp.test.js

```diff
@@ -0,0 +1,117 @@
+import { screen } from '@testing-library/react';
+import React from 'react';
+
+import { EditCartItemPopUp } from '../../../../components/CartContent/EditPopUp/EditCartItemPopUp';
+import { LOCAL_STORAGE_CART_KEY } from '../../../../globals';
+import { mockDepositCart, mockWithdrawCart } from '../../../../mocks/cart';
+import { render, userEvent } from '../../../../testSetup';
+
+describe('<EditCartItemPopUp />', () => {
+  beforeEach(() => {
+    setItemSpy = jest.fn();
+    global.localStorage.__proto__.setItem = setItemSpy;
+    global.localStorage.__proto__.getItem = jest.fn(() =>
+      JSON.stringify(mockDepositCart),
+    );
+  });
+
+  afterEach(() => {
+    setItemSpy.mockClear();
+  });
+  it('renders correctly', () => {
+    const mockCartItem = mockDepositCart[0];
+    render(
+      <EditCartItemPopUp
+        cartItem={mockCartItem}
+        open={true}
+        handleClose={jest.fn()}
+      />,
+    );
+
+    expect(screen.getByText(mockCartItem.name)).toBeInTheDocument();
+    expect(screen.getByLabelText('Expiry Date')).toHaveValue(
+      mockCartItem.expiry_dates[0].expiry_date ?? 'No Expiry',
+    );
```

Comment:
Add a test case when the expiry_date property is non-null, and should be displayed